src/vs/workbench/contrib/notebook/common/notebookCommon.ts

```diff
@@ -646,22 +646,32 @@ export namespace CellUri {
 		});
 	}
 
-	export function parseCellOutputUri(uri: URI): { notebook: URI; outputId?: string } | undefined {
+	export function parseCellOutputUri(uri: URI): { notebook: URI; openIn: string; outputId?: string; cellFragment?: string; outputIndex?: number } | undefined {
 		if (uri.scheme !== Schemas.vscodeNotebookCellOutput) {
 			return;
 		}
 
 		const params = new URLSearchParams(uri.query);
-		const outputId = params.get('outputId') || undefined;
+		const openIn = params.get('openIn');
+		if (!openIn) {
+			return;
+		}
+		const outputId = params.get('outputId') ?? undefined;
+		const cellFragment = uri.fragment ?? undefined;
+		const outputIndex = params.get('outputIndex') ? parseInt(params.get('outputIndex') || '', 10) : undefined;
```

Comment:
How can we have a Cell Output Uri that doesn't contain `outputId`, `cellFragment` and `outputIndex`?
Woudln't that be an invalid Cell Output Uri?
Or are you adding this code for backwards compatibility for some old Uris that may linger from a few days ago?
I would have thought they are all non-optional and must have valid values

Reply:
backwards compatible since we still use Ids for vscode dev cases